### PR TITLE
Fix removing commas from function arguments in maps

### LINKED ISSUE #13984: Regression in 2.8.1: Commas are removed from fn args in Sass maps.
Probably caused by https://github.com/prettier/prettier/pull/9184?
Possible related to https://github.com/prettier/prettier/issues/13580?

**Prettier 2.8.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABOBDMALABAEigwFs4AaA7AVzlwEspKaBnXYAHUYOIwAclcPXgDpSAJwDmcABSdc87nzJyFsrgtwByAMTB8vMXABmdAB4BfALS7CJOOc0CARgBsEAE2HM4MaficY3hT41HACITQAlMrquJEqkQDcnOacIGQgELwwdNDMyKAYYmIQAO4ACkUI+SgYLqUYAJ75GU5iWADWPgDKdgAyDHDIRnVBIG2dPbxYDBLIMGI0GXDETnDu7ut9GFASVBhSAGIQYjwwObvIIBhUMBDpINgwxC4A6th08MzTYHDd1Z86AA3T6NK5gZgtEAMbxiGDldoSHjDUbkEAAK2Ypm6szcAEUqBB4CiXGNpmJYVdmBCoQYGDBXnR3DBsMgABwABgyBgg3le7V4VwMcFhQKGGQAjoT4AisjVrsxLFA4Ot1g9DFK6IYEQdkUgRqS0d5iHR5osjbi4ABBc5iOhOW5wcpwMQDZUksbMS0EolDfWojIwDBORnM1lIABMgfadBcswAwhBiHqQCKAKwPKjeAAqwZqBrGQJoAEkoJtYN0wHbslay90YI03B77OYgA)

```sh
--parser scss
```

**Input:**

```scss
@each $name, $hue in $hues {
  $map: map.merge(
    $map,
    (
      '#{$prefix}-#{$name}': blend.set($base, $hue: $hue),
    )
  );
}

```

**Output:**

```scss
@each $name, $hue in $hues {
  $map: map.merge($map, ('#{$prefix}-#{$name}': blend.set($base $hue: $hue)));
}

```

**Expected behavior:**

Comma should not be removed from `blend.set($base, $hue: $hue)` call.

### LINKED ISSUE #14088: Idempotent issue for SCSS map
**Prettier 2.8.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASAJgSwM4AcA2AhgJ4C0ARgE5yEDWuEms2SABALaG6npxzfs4lAOZwAFAB0orVpOkzWEkAA9spaPmJK2SjcVbYw1BK0JR0szstIB3TOhgALNgGJgnbqJhjUwyvYpqOgYmGGwAGkUQbHYlAEpSAEYAXzilcKkZOIz5DBwCEkCaekZmKTiAbhBwkAhcGExobGRQQkpKCBsABTaEZpRCfBsSZpqqQjBaOBgAZUJBABkmOGQAM0HsODHKCanZ3AmmYWQYSgBXLZA4dnI4dF50BbNhM8JRADEISk4YBqhjgZnGAQaogRwwdj4ADqjkw8DwEzgMz6cMwADc4cRkOBsKMQExNpQYF0dsJOGsNpcAFbYZQzI74OAARTOEHgFPwmxqB0ohOxhlxoNw-lgUPsTmQAA4AAzcjqbKE7XDY4VwQlolY1ACOrPgJLq-RAhDUUD4D1B1B1mGoJLe5KQ605l027EwJ3OzoZcAAgr9-OQgXAukIlqaOVzol6WWyVg7KTUYIRyGKHI5kAAmBM7TD4I4AYQg7HtV2wAFZQWdNgAVJP9R0RtEXACS5gQsyMmHq3vMMxgxEZ4bgyWSQA)

```sh
--parser scss
```

**Input:**

```scss
$display-breakpoints: map-deep-merge(
  (
    "xs-only": "only screen and (max-width: #{map-get($grid-breakpoints, "sm")-1})",
  ),
  $display-breakpoints
);
```

**Output:**

```scss
$display-breakpoints: map-deep-merge(
  (
    "xs-only": "only screen and (max-width: #{map-get($grid-breakpoints, "sm")-1})"
  ),
  $display-breakpoints
);

```

**Second Output:**

```scss
$display-breakpoints: map-deep-merge(
  (
    "xs-only": "only screen and (max-width: #{map-get($grid-breakpoints, " sm
      ")-1})",
  ),
  $display-breakpoints
);

```

**Expected behavior:**

same as input

**Related:**

- https://github.com/prettier/prettier/issues/9128
- https://github.com/prettier/prettier/issues/13984

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #13984
Fixes #14088
Closes #13995

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-css/printer-postcss.js [hunk 1]
```diff
@@ -853,6 +856,20 @@ function genericPrint(path, options, print) {
           continue;
         }
 
+        if (isMapItemNode) {
+          const isPartOfValue =
+            node.groups?.[1]?.type === "value-colon" && i > 1;
+          if (
```
Comment (fisker) [inline src/language-css/printer-postcss.js:862]: Can you add an example here to show what this condition supposed to match?

**Review by fisker — APPROVED** [review]
Not sure how this works, but fine to me since it doesn't break things and fix the new issue.

File: src/language-css/printer-postcss.js [hunk 2]
```diff
@@ -857,6 +857,13 @@ function genericPrint(path, options, print) {
...
 
         if (isMapItemNode) {
+          // Nodes after the colon are considered to be the value part of a key-value pair
+          //   $foo: map-fn(
+          //     (
+          //       "foo"                 :                 "bar #{inner-fn($baz)}"
+          //       ^^^^^(node.groups[0]) ^(node.groups[1]) ^^^^^^^^^^^^^^^^^^^^^^^(isPartOfValue === true)
+          //     )
+          //   );
           const isPartOfValue =
```
Comment (fisker) [inline src/language-css/printer-postcss.js:867]: isPartOfValue only used once, should be able to use one `if`

File: src/language-css/printer-postcss.js (hunk 1 above)
Comment (fisker) [reply src/language-css/printer-postcss.js:869]: Actually, I mean example for the code in this condition, not sure those `value-string`/`value-word` suppose to match.

**Comment by fisker** [comment]
https://github.com/prettier/prettier/pull/14089#discussion_r1059512974

File: src/language-css/printer-postcss.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-css/printer-postcss.js:869]: What do you think the following comment?

```js
// For example, there is the following key-value pair:
//
//   "xs-only": "only screen and (max-width: #{map-get($grid-breakpoints, "sm")-1})"
//
// "only screen and (max-width: #{map-get($grid-breakpoints, " is a "value-string"
// and "sm" is a "value-word"
// We should not insert any spaces and lines here.
```

**Comment by fisker** [comment]
https://github.com/prettier/prettier/pull/14089#discussion_r1059512627

**Comment by sosukesuzuki** [comment]
@fisker Why assigned to you? Are you working on this? or reviewing?

**Comment by fisker** [comment]
**Prettier pr-14089**
[Playground link](https://deploy-preview-14089--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEASAJgSwM4AcA2AhgJ4C0ARgE5yEDWuEms2SABALaG6npxzfs4lAOZwAFAB0orVpOkzWEkAA9spaPmJK2SjcVbYw1BK0JR0szstIB3TOhgALNgGJgnbqJhjUwyvYpqOgYmGGwAGkUQJQBKUgBGAF8YpXCpGRi0+QwcAhJAmnpGZikYgG4QcJAIXBhMaGxkUEJKSggbAAUWhEaUQnwbEkaqqkIwWjgYAGVCQQAZJjhkADN+7DgRyjGJ6dwxpmFkGEoAVw2QOHZyOHRedDmzYRPCUQAxCEpOGDqoQ76TmAQSogRwwdj4ADqjkw8DwYzgUx6MMwADcYcRkOBsMMQEx1pQYB0tsJOCs1ucAFbYZRTA74OAARROEHgZPw6yqe0o+MxhmxwNw-lgEPsTmQAA4AAyctrrCFbXCYwVwfEopZVACOzPgRJqvRAhDUUD4d2B1C1mGoRJepKQq3Z53W7EwR1OjrpcAAgt9-OQAXAOkIFsa2RyQNgPUyWUs7eSqjBCOQRQ5HMgAEzxraYfAHADCEHYtou2AArMCTusACqJ3r2sMos4ASXMCGmRkwtU95imMGI9NDcESiSAA)

```sh
--parser scss
```

**Input:**

```scss
$display-breakpoints: map-deep-merge(
  (
    "xs-only": "only screen and (max-width: #{map-get($grid-breakpoints, "")-1})",
  ),
  $display-breakpoints
);
```

**Output:**

```scss
$display-breakpoints: map-deep-merge(
  (
    "xs-only": "only screen and (max-width: #{map-get($grid-breakpoints, "
      ")-1})",
  ),
  $display-breakpoints
);

```

**Comment by fisker** [comment]
I'm taking a deeper look.

**Comment by fisker** [comment]
@sosukesuzuki  [`29842ce`](https://github.com/prettier/prettier/pull/14089/commits/29842ce7f723a83f972648868d73cd383d0d0dce) should be a better fix. But still not right.

```scss
"sm-only": "min-width: #{call(" + 1 + 2 + ")}",
```

Anything can be inside the sting.

**Comment by fisker** [comment]
What do you think of current version?

**Comment by sosukesuzuki** [comment]
@fisker Looks good to me, thank you.

File: src/language-css/utils/index.js [hunk 3]
```diff
@@ -276,7 +276,8 @@ function isSCSSMapItemNode(path) {
   // Check open parens contain key/value pair (i.e. `(key: value)` and `(key: (value, other-value)`)
   if (
     !isKeyValuePairInParenGroupNode(node) &&
-    !(parentParentNode && isKeyValuePairInParenGroupNode(parentParentNode))
+    !(parentParentNode && isKeyValuePairInParenGroupNode(parentParentNode)) &&
+    !isKeyValuePairNode(node)
```
Comment (fisker) [inline src/language-css/utils/index.js:280]: Should we revert this line?

Comment (sosukesuzuki) [reply src/language-css/utils/index.js]: Thank you! [ee114cb](https://github.com/prettier/prettier/pull/14089/commits/ee114cbcc708654cf063593356602da8ebf18e19)

**Review by fisker — APPROVED** [review]

**Pull request merged**